app/models/concerns/host_common.rb

```diff
@@ -204,6 +204,24 @@ def available_puppetclasses
     environment.puppetclasses - parent_classes
   end
 
+  def inherited_params(include_source = false)
+    hp = {}
+    params = host_inherited_params_objects
+    params.each do |param|
+      case param
+        when CommonParameter
+          hp.update(Hash[param.name => include_source ? {:value => param.value, :source => N_('common').to_sym, :safe_value => param.safe_value} : param.value])
+        when GroupParameter
+          hp.update(Hash[param.name => include_source ? {:value => param.value, :source => N_('hostgroup').to_sym, :safe_value => param.safe_value, :source_name => hostgroup.title} : param.value])
+        else
+          source = param.class.to_s.gsub('Parameter', '').downcase
+          source = 'operatingsystem' if source == 'os'
```

Comment:
I moved the method here as is so before I move it back and loose the comment - I can change this line for operating system but what do I do with line 217? do I check the other 3 options here?